cmd/bosun/conf/system.go

```diff
@@ -0,0 +1,406 @@
+// EnabledBackends returns and EnabledBackends struct which contains fields
+// to state if a backend is enabled in the configuration or not
+func (sc *SystemConf) EnabledBackends() EnabledBackends {
+	b := EnabledBackends{}
+	b.OpenTSDB = sc.OpenTSDBConf.Host != ""
+	b.Graphite = sc.GraphiteConf.Host != ""
+	b.Influx = sc.InfluxConf.URL.URL != nil && sc.InfluxConf.URL.Host != ""
+	b.Logstash = len(sc.LogStashConf.Hosts) != 0
+	b.Elastic = len(sc.ElasticConf.Hosts) != 0
+	return b
+}
+
+// OpenTSDBConf contains OpenTSDB specific configuration information. The ResponseLimit
+// will prevent Bosun from loading responses larger than its size in bytes. The version
+// enables certain features of OpenTSDB querying
+type OpenTSDBConf struct {
+	ResponseLimit int64
+	Host          string           // OpenTSDB relay and query destination: ny-devtsdb04:4242
+	Version       opentsdb.Version // If set to 2.2 , enable passthrough of wildcards and filters, and add support for groupby
+}
+
+// GraphiteConf contains a string representing the host of a graphite server and
+// a map of headers to be sent with each Graphite request
+type GraphiteConf struct {
+	Host    string
+	Headers map[string]string
+}
+
+// AnnotateConf contains the elastic configuration to enable Annotations support
+type AnnotateConf struct {
+	Hosts []string // CSV of Elastic Hosts, currently the only backend in annotate
+	Index string   // name of index / table
+}
+
+// LogstashConf contains a list of elastic hosts for the depcrecated logstash functions
+type LogStashConf struct {
+	Hosts expr.LogstashElasticHosts
+}
+
+// ElasticConf contains configuration for an elastic host that Bosun can query
+type ElasticConf struct {
+	Hosts expr.ElasticHosts
+}
+
+// ElasticConf contains configuration for an influx host that Bosun can query
+type InfluxConf struct {
+	URL       URL
+	Username  string
+	Password  string `json:"-"`
+	UserAgent string
+	Timeout   Duration
+	UnsafeSSL bool
+	Precision string
+}
+
+// DBConf stores the connection information for Bosun's internal storage
+type DBConf struct {
+	RedisHost     string
+	RedisDb       int
+	RedisPassword string
+
+	LedisDir      string
+	LedisBindAddr string
+}
+
+// SMTPConf contains information for the mail server for which bosun will
+// send emails through
+type SMTPConf struct {
+	EmailFrom string
+	Host      string
+	Username  string
+	Password  string `json:"-"`
+}
+
+// GetSystemConfProvider returns the SystemConfProvider interface
+// and validates the logic of the configuration. If the configuration
+// is not valid an error is returned
+func (sc *SystemConf) GetSystemConfProvider() (SystemConfProvider, error) {
+	var provider SystemConfProvider = sc
+	if err := ValidateSystemConf(sc); err != nil {
+		return provider, err
+	}
+	return provider, nil
+}
+
+// NewSystemConf retruns a system conf with default values set
+func newSystemConf() *SystemConf {
+	return &SystemConf{
+		CheckFrequency:  Duration{Duration: time.Minute * 5},
```

Comment:
That was my original thought as well. The reason I put them in system was more of a permissions. This is because editing those will change the way *all* alerts in Bosun will run. So having those in the system conf seemed safer to me.